src/Layer.js

```diff
@@ -446,11 +454,11 @@ export default class Layer {
         this._renderWaiters.forEach(resolve => resolve());
 
         if (this.isAnimated()) {
-            this._needRefresh();
-        }
-
-        if (this._state !== states.INIT) {
-            this._fire('updated');
+            this._needRefresh().then(() => {
+                if (this.isPlaying()) {
+                    this._fire('updated'); // this should be restricted to animation.isPlaying?
```

Comment:
What do you mean with the comment? (`this should be restricted to animation.isPlaying?`): to only fire update here if `this` is an Animation and isPlaying? (in that case since all this executes only if the Layer `isAnimated()` I see no further need to check `this`)

Reply:
The comment was old, before adding isPlaying.... Now it is not relevant, but the if(isPlaying)... it is!